subsys/bluetooth/scan.c

```diff
@@ -283,6 +319,141 @@ static void notify_connecting_error(struct bt_scan_device_info *device_info)
 	}
 }
 
+static bool blacklist_device_check(const bt_addr_le_t *addr)
+{
+	bool blacklist_device = false;
+
+	k_mutex_lock(&blacklist_mutex, K_FOREVER);
+
+	for (size_t i = 0; i < bt_scan.blacklist.cnt; i++) {
+		if (bt_addr_le_cmp(&bt_scan.blacklist.addr[i], addr) == 0) {
+			blacklist_device = true;
+		}
+	}
+
+	k_mutex_unlock(&blacklist_mutex);
+
+	return blacklist_device;
+}
+
+static void attempts_filter_force_add(struct conn_attempts_filter *filter,
+				      const bt_addr_le_t *addr)
+{
+	/* Overwrite the oldest device */
+	memset(filter->oldest_device, 0, sizeof(*filter->oldest_device));
```

Comment:
We may simply set the `attempts` to zero. We no longer need the memset

Reply:
Agree, that seems better. The address is overwritten below anyway.

Reply:
Done